<launch>
  <!-- image_raw, camera_info -> image_rect -> ImageFeature0D -> ObjectDetection -->
  <!-- And compare ObjectDetection with TF from rosbag -->

  <arg name="ROBOT" default="sim"/>
  <include file="$(find pr2_machine)/$(arg ROBOT).machine"/>

  <param name="/use_sim_time" value="true"/>

  <!-- inputs are tf, image_raw, camera_info, joint_states -->
  <node name="rosbag_play"
        pkg="rosbag" type="play"
        args="$(find elevator_move_base_pr2)/test/test-panel-pose-detection.bag -r 0.2 --clock --loop"/>

  <group ns="/narrow_stereo/left" >
    <node name="image_proc"
          pkg="image_proc" type="image_proc">
      <rosparam>
        queue_size: 100  <!-- image is throttled ?? -->
      </rosparam>
    </node>
  </group>

  <!-- Throttle image -->
  <group ns="/narrow_stereo/left">
    <node name="throttle"
          pkg="jsk_topic_tools" type="lightweight_throttle"
          machine="c1">
      <remap from="~input" to="image_rect"/>
      <remap from="~output" to="image_rect_throttle"/>
      <rosparam>
        update_rate: 2.0
      </rosparam>
    </node>
  </group>

  <!-- Detect elevator_call_panel / elevator_inside_panel. -->
  <group ns="/narrow_stereo/left">
    <node name="imagesift"
          pkg="nodelet" type="nodelet"
          args="standalone imagesift/ImageSift"
          machine="c2"
          launch-prefix="nice -n +10">
      <remap from="image" to="image_rect_throttle"/>
    </node>
    <include file="$(find elevator_move_base_pr2)/launch/elevator_panels_detection_eng2.launch">
      <arg name="INPUT_FEATURE" value="ImageFeature0D"/>
      <arg name="MACHINE" value="c2"/>
    </include>
    <group ns="/narrow_stereo/left/point_pose_extractor_elevator_call_panel">
      <rosparam>
        error_threshold: 500
      </rosparam>
    </group>
  </group>

  <!-- test for panel pose detection -->
  <test test-name="test_panel_pose_detection"
        name="test_panel_pose_detection"
        pkg="elevator_move_base_pr2" type="test-panel-pose-detection.l"
        time-limit="300">
    <rosparam>
      max_distance: 100
    </rosparam>
  </test>

</launch>
